providers/apis/defi/ethmulticlient/client.go

```diff
@@ -100,9 +100,9 @@ func (c *GoEthereumClientImpl) BatchCallContext(ctx context.Context, calls []rpc
 
 	if err = c.client.BatchCallContext(ctx, calls); err != nil {
 		c.apiMetrics.AddRPCStatusCode(c.api.Name, c.redactedURL, metrics.RPCCodeError)
-		return
+		return err
```

Comment:
I assume this was done for clarity since it can be confusing when we have named return values. Consider removing the name entirely.

(i.e., instead of `(err error)` just let it be `error` for the return type.

Reply:
https://github.com/skip-mev/connect/pull/797/commits/98c6a2b12b1805de36a769137fb6cc108acf488a